extensions/amp-video/0.1/video-cache.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../src/services';
+import {createElementWithAttributes} from '../../../src/dom';
+import {toArray} from '../../../src/types';
+
+/** @private {?boolean} store the result of whether the document can use cache url */
+let docSupportsCacheResult = null;
+
+/**
+ * Add the caching sources to the video if opted in.
+ * @param {!AmpVideo} video
+ * @return {!Promise}
+ */
+export function addCacheSources(video) {
+  if (!docSupportsCache(video.win)) {
+    return Promise.resolve();
+  }
+  const servicePromise = Services.cacheUrlServicePromiseForDoc(video.element);
+  return servicePromise
+    .then((service) =>
+      Promise.all([
+        service.createCacheUrl(selectVideoSource(video.element)),
+        Services.storyRequestServiceForOrNull(video.win),
+      ])
+    )
+    .then((promiseResult) => {
+      const requestUrl = promiseResult[0].replace('/c/', '/mbv/');
```

Comment:
This is not how the URL has to be formatted, does it work?

the expected is: `https://example-com.cdn.ampproject.org/mbv/(if SSL: s)/{videosourceurl}`

Reply:
Yes, the url returned by the cacheUrl service is exactly like that but has the `/c` instead of the `/mbv` (it automatically adds the `/s`) so after the cacheUrl transformation, we just have to replace that part and we're good. Since the URL is a video, it will always have a `/c` at the beginning of the path from the amp toolbox.